### ISSUE TITLE
ld: library not found for -lclang.dy

### ISSUE THREAD

**Issue by mephi42** [KyleMayes/clang-sys#84]
I'm trying to use clang-sys on macOS as follows:

```
clang-sys = { version = "0.24.0", features = ["clang_6_0"] }
```

The build fails with:

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-L" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.1y16o1qfye96o7m0.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.2339ye2qsut5jzf1.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.2xs38k3nu4oe9uo6.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.3fiyi1r0f9gw01la.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.3rngp6bm2u2q5z0y.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.4011gx3bwpzq03se.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.4oc10dk278mpk1vy.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.4xq48u46a1pwiqn7.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.4ypvbwho0bu5tnww.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.8xzrsc1ux72v29j.rcgu.o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.oa3rad818d8sgn4.rcgu.o" "-o" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135" "$PROJECT/target/debug/deps/test_main-f36430e3bd34d135.crate.allocator.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "$PROJECT/target/debug/deps" "-L" "/usr/local/opt/llvm/lib" "-L" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libtest-8b11cd23e28267c4.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libterm-97c75d65953833c3.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libgetopts-96e8d9c41fd5f863.rlib" "$PROJECT/target/debug/deps/$PROJECT-9d25e484d92983e9.rlib" "$PROJECT/target/debug/deps/libclang_sys-7109b6c4a521deab.rlib" "$PROJECT/target/debug/deps/liblibc-aecf7ee5e4c12d5c.rlib" "$PROJECT/target/debug/deps/libglob-186622082d7523d5.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-90d0baaa628316fc.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-2501f1a43ee00253.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc_jemalloc-1ebc0af87eedc756.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libunwind-d5a51010c907c42d.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc_system-776df9543761d8dc.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liblibc-56ad06cec3622785.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc-54ae7b5ec4f1ba32.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcore-585d1da946bc8c2f.rlib" "$HOME/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-a06451ee7f3e01cf.rlib" "-l" "clang.dy" "-l" "System" "-l" "resolv" "-l" "pthread" "-l" "c" "-l" "m"
  = note: ld: library not found for -lclang.dy
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Somehow it tries to use `libclang.dy`, and not `libclang.dylib`.

Also, v0.23.0 compiles just fine.

**Comment by KyleMayes** [comment]
Thanks for the bug report.
I have determined the cause and will have a fix soon.

**Comment by KyleMayes** [comment]
Could you try the fix in the `macos` branch?
Thanks.

**Comment by mephi42** [comment]
Works, thanks!

**Comment by KyleMayes** [comment]
This fix has been released as `v0.26.2`.
